mycpp/control_flow_pass.py

```diff
@@ -130,6 +145,84 @@ def resolve_callee(self, o: CallExpr) -> Optional[util.SymbolPath]:
         # Don't currently get here
         raise AssertionError()
 
+    def get_variable_name(self, expr: Expression) -> Optional[util.SymbolPath]:
+        """
+        To do dataflow analysis we need to track changes to objects, which
```

Comment:
Is there a term we can use besides "name"?

This is basically an identifier for expressions?

why not "expr_id" or something?

or "cfg_expr_id" ?


or "flow_expr_id" ?   (to avoid "CFG == context free grammar"  problem :) )

Reply:
Hm what's wrong with "name"? I think it's accurate, no? We aren't trying to name every expression per se. Most expressions, e.g. `'foo'` or `{}` or `[1, 2, 3]` won't have a name at all

Reply:
Hmmm ... well "variable name" is something like `x`

This is also giving a "name" `t.0` to an expression t[0] .   That's a slightly different thing than a variable name

and `module.SomeObject.member_a`

Maybe it could be "ID for reference to a value" or  "name of reference" or "name of value"

I think what it is getting at is that

```
t= (42, 2, 3, 4)
x = t[0]
y = x
z = t[2-2]

```

there is one VALUE 42 which has a bunch of different ways to refer to it, and the ways need some kind of ID , right?   

to understand aliasing

----


But I guess we can think about this later, maybe it will be more clear with the rest of the code